### PR TITLE
10.0 project invoicing

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: project_invoicing/models/account_analytic_line.py [hunk 1]
```diff
@@ -16,36 +16,57 @@ class AccountAnalyticLine(models.Model):
...
-            else:
-                line.partner_invoice_id = None
+    @api.model
+    def create(self, vals):
+        line = super(AccountAnalyticLine, self).create(vals)
+        if line.project_id:
+            line.partner_invoice_id = line.project_id.partner_id
+        return line
+
+    @api.model
```
Comment (joaoalf) [inline project_invoicing/models/account_analytic_line.py]: @dufresnedavid I think that the decorator here is `api.multi`

File: project_invoicing/models/project_project.py [hunk 2]
```diff
@@ -19,12 +19,22 @@ class ProjectProject(models.Model):
     def generate_invoices(self, data):
         self.ensure_one()
 
-        lines_by_partner = defaultdict(list)
+        self = self.with_context(
+            force_company=self.company_id.id,
```
Comment (joaoalf) [inline project_invoicing/models/project_project.py:23]: @dufresnedavid Why `company_id` and `force_company_id`?

**Review by joaoalf — CHANGES_REQUESTED** [review]

File: project_invoicing/models/project_project.py (hunk 2 above)
Comment (ghost) [reply project_invoicing/models/project_project.py:23]: to ensure that the correct values are taken for multi-company properties used (on partners and products) when generating the invoice. This ensures that a user with multi company access may trigger an invoice from a project related to another company.

File: project_invoicing/models/account_analytic_line.py (hunk 1 above)
Comment (ghost) [reply project_invoicing/models/account_analytic_line.py]: @joaoalf right

File: project_invoicing/models/project_project.py (hunk 2 above)
Comment (ghost) [reply project_invoicing/models/project_project.py:23]: Also, force_company and company_id are both used by odoo at different places.

**Review by joaoalf — APPROVED** [review]

**Pull request merged**